test/autorest/custombaseurlgroup/zz_generated_paths_client.go

```diff
@@ -20,12 +21,24 @@ import (
 // PathsClient contains the methods for the Paths group.
 // Don't use this type directly, use NewPathsClient() instead.
 type PathsClient struct {
-	con *Connection
+	host string
+	pl   runtime.Pipeline
 }
 
 // NewPathsClient creates a new instance of PathsClient with the specified values.
-func NewPathsClient(con *Connection) *PathsClient {
-	return &PathsClient{con: con}
+func NewPathsClient(host *string, options *azcore.ClientOptions) *PathsClient {
```

Comment:
This is an oddity I wanted to discuss.  The swagger defines this as required, however it has a default value, so we consider it optional.  This is how we used to generate the Connection params, however I wonder if in this case we should actually generate a `PathClientOptions` that embeds the `azcore.ClientOptions` and contains the `host` param.  It's more work in the code generator, but I feel like it's the correct thing to do.  Thoughts?

Reply:
Moved optional params to an options struct for non-data plane scenarios.